builtin/core.wren

```diff
@@ -21,6 +21,16 @@ class Sequence {
     }
     return true
   }
+
+  reduce(acc, f) {
+    for (element in this) {
+      acc = f.call(acc, element)
+    }
+    return acc
+  }
+
+  reduce(f) { this[1..-1].reduce(this[0], f) }
```

Comment:
> Do all sequences have indexing?

No, I don't think `Sequence` should assume that.

> Sure, right now List is the only implementing class, but...

Range does too. Maybe String?